WcaOnRails/lib/results_validators/persons_validator.rb

```diff
@@ -65,17 +65,17 @@ def validate(competition_ids: [], model: Result, results: nil)
         end
 
         without_wca_id, with_wca_id = persons_by_id.values.partition { |p| p.wca_id.empty? }
-        if without_wca_id.any?
-          existing_person_in_db_by_name = Person.where(name: without_wca_id.map(&:name)).group_by(&:name)
-          existing_person_in_db_by_name.each do |name, persons|
-            @warnings << ValidationWarning.new(:persons, competition_id,
-                                               SAME_PERSON_NAME_WARNING,
-                                               name: name,
-                                               wca_ids: persons.map(&:wca_id).join(", "))
-          end
-        end
         duplicate_newcomer_names = []
         without_wca_id.each do |p|
+          existing_person = Person.where("LOWER(name) = ?", p.name.downcase).group_by(&:name)
```

Comment:
Thanks a lot for your PR!
In my opinion something needs to be changed here: there used to be one query for all newcomers, which were then grouped by name.
Now there is one query per newcomer, and if I'm understanding this right there is no need for the group_by since the name is known.

Can I suggest to try avoiding doing N queries and keep only one, for instance by doing something like this:
```Person.where("LOWER(name) IN (?)", without_wca_id.map(&:name).map(&:downcase)).group_by(&:name)```?
I believe it retains the same behavior while ignoring the case.

Reply:
Yup, worked great. Have done basically 0 work with SQL before so I did not think about IN existing.